### PR TITLE
Added user authentication

### PR DESCRIPTION
**Pull request by melissadi**
- Added new fields to user sign up form
- Added new columns to User table
- Updated validations in User model (and created STATES constant for `home_state` validation)
- Updated sign-in and sign-up tests
- Updated FactoryBot
- Updated Rspec test for User model

### CODE REVIEW

File: app/models/constants.rb [hunk 1]
```diff
@@ -0,0 +1 @@
+STATES = ["AK","AL","AR","AS","AZ","CA","CO","CT","DC","DE","FL","GA","GU","HI","IA","ID","IL","IN","KS","KY","LA","MA","MD","ME","MI","MN","MO","MS","MT","NC","ND","NE","NH","NJ","NM","NV","NY","OH","OK","OR","PA","PR","RI","SC","SD","TN","TX","UT","VA","VI","VT","WA","WI","WV","WY"]
```
Comment (CassandraNewell) [inline app/models/constants.rb]: Please move this file to `app/constants/states.rb`

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -1,5 +1,15 @@
 require 'rails_helper'
 
 RSpec.describe User, type: :model do
```
Comment (CassandraNewell) [inline spec/models/user_spec.rb]: Please test for last name as well -- using shoulda-matchers will make this easier :)

**Comment by melissadi** [comment]
- Updated User model test (now using shoulda-matchers)
- Updated User model validation
- Resolved merge conflicts

File: app/models/constants/states.rb [hunk 3]
```diff
@@ -0,0 +1 @@
+STATES = ["AK","AL","AR","AS","AZ","CA","CO","CT","DC","DE","FL","GA","GU","HI","IA","ID","IL","IN","KS","KY","LA","MA","MD","ME","MI","MN","MO","MS","MT","NC","ND","NE","NH","NJ","NM","NV","NY","OH","OK","OR","PA","PR","RI","SC","SD","TN","TX","UT","VA","VI","VT","WA","WI","WV","WY"]
```
Comment (CassandraNewell) [inline app/models/constants/states.rb]: Ack, I'm sorry, yesterday Casi was not thinking clearly. These constants should actually be part of your `State` class, so they belong in `/app/models/state.rb`. You should then be able to access them elsewhere as `State::STATES`.

**Comment by melissadi** [comment]
- Followed Casi's changes (moved STATES into State model and updated User model test accordingly)

**Comment by melissadi** [comment]
- Resolved merge conflicts

**Pull request merged**